xapian-letor/tests/api_letor.cc

```diff
@@ -192,40 +241,106 @@ DEFINE_TESTCASE(createfeaturevectoronevector, generated)
 DEFINE_TESTCASE(createfeaturevectoronevector_wrongquery, generated)
 {
     Xapian::FeatureList fl;
-    Xapian::Database db = get_database("apitest_ranker3",
+    Xapian::Database db = get_database("apitest_wrongquery",
 				       db_index_one_document);
```

Comment:
It's good to clean up these database names, but for generated databases it's better to align the names with the generator functions rather than with the testcase names.

For example, currently the generator function `db_index_one_document` is used to build 11 differently named read-only databases, but this generator function is not parameterised so all these databases will have identical contents.  If we used the same name in each of the 11 testcases, then only one database would need to be built for the first testcase to run, and that would then be reused for the other 10 testcases.

Reply:
ok I will revert this change.

Reply:
My point is not that the existing `apitest_ranker3` is right, rather that a name based on the generator function would be better - e.g. `apitest_index_one_document`.  If each testcase which uses this generator function uses that same name, then we get the speed and disk space savings from reusing the cached database.

Reply:
got it.